### PR TITLE
feat(new-contracts): update emp launch to use new emp contract

### PR DESCRIPTION
**Pull request by pemulis**
Signed-off-by: John Shutt <john.d.shutt@gmail.com>

This updates the EMP Launch repo to use the new EMP contract, and makes a few other minor improvements.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -43,13 +43,13 @@ if (argv.gasprice < 1 || argv.gasprice > 1000) throw "--gasprice must be between
...
-    sponsorDisputeRewardPct: { rawValue: toWei("0.05") }, // 5% reward for sponsors who are disputed invalidly
-    disputerDisputeRewardPct: { rawValue: toWei("0.2") }, // 20% reward for correct disputes.
+    disputeBondPercentage: { rawValue: toWei("0.1") }, // 10% dispute bond.
+    sponsorDisputeRewardPercentage: { rawValue: toWei("0.05") }, // 5% reward for sponsors who are disputed invalidly
+    disputerDisputeRewardPercentage: { rawValue: toWei("0.2") }, // 20% reward for correct disputes.
     minSponsorTokens: { rawValue: toWei("100") }, // Min sponsor position size of 100 synthetic tokens.
     liquidationLiveness: 7200, // 2 hour liquidation liveness.
     withdrawalLiveness: 7200, // 2 hour withdrawal liveness.
-    excessTokenBeneficiary:  getAddress("Store", networkId), // UMA Store contract.
+    financialProductLibraryAddress:  getAddress("ExpiringMultiPartyLib", networkId), // ExpiringMultiParty library contract.
```
Comment (smb2796) [inline index.js:52]: Do you think this should also be a passed arg? Or should it almost always default to ExpiringMultiPartyLib?

File: index.js [hunk 2]
```diff
@@ -72,6 +72,7 @@ if (argv.gasprice < 1 || argv.gasprice > 1000) throw "--gasprice must be between
   // Since the simulated transaction succeeded, send the real one to the network.
   const { transactionHash } = await empCreator.methods.createExpiringMultiParty(empParams).send(transactionOptions);
   console.log("Deployed in transaction:", transactionHash);
+  process.exit(0);
```
Comment (smb2796) [inline index.js:75]: Good call on this. Was not optimal that previously it would just hang

File: index.js (hunk 1 above)
Comment (pemulis) [reply index.js:52]: That's a good point. We could allow this to be an argument passed in and default to `ExpiringMultiPartyLib` if there isn't one.

File: index.js [hunk 3]
```diff
@@ -59,12 +60,12 @@ if (argv.gasprice < 1 || argv.gasprice > 1000) throw "--gasprice must be between
     syntheticSymbol: argv.syntheticSymbol, // Short name.
     collateralRequirement: { rawValue: toWei("1.25") }, // 125% collateral req.
     disputeBondPercentage: { rawValue: toWei("0.1") }, // 10% dispute bond.
-    sponsorDisputeRewardPercentage: { rawValue: toWei("0.05") }, // 5% reward for sponsors who are disputed invalidly.
+    sponsorDisputeRewardPercentage: { rawValue: toWei("0.05") }, // 5% reward for sponsors who are disputed invalidly
     disputerDisputeRewardPercentage: { rawValue: toWei("0.2") }, // 20% reward for correct disputes.
-    minSponsorTokens: { rawValue: toWei(argv.minSponsorTokens.toString()) }, // Minimum sponsor position size.
+    minSponsorTokens: { rawValue: toWei("100") }, // Min sponsor position size of 100 synthetic tokens.
```
Comment (smb2796) [inline index.js:65]: I think this param should be passed in as an arg. 100 synths could vary wildly

**Review by smb2796 — APPROVED** [review]
Looks good to me except I think that minSynths should be passed as an arg instead of being hardcoded. What do you think?

**Comment by pemulis** [comment]
> Looks good to me except I think that minSynths should be passed as an arg instead of being hardcoded. What do you think?

Sounds good! I'll make that change and merge.

**Pull request merged**